components/movementsensor/adxl345/adxl345.go

```diff
@@ -232,6 +331,123 @@ func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error
 	return handle.WriteByteData(ctx, register, value)
 }
 
+func getByteFromInterrupts(interrupts []string) byte {
+	var register byte
+	for _, interrupt := range interrupts {
+		register += interruptBitPosition[interrupt]
+	}
+	return register
+}
+
+func (adxl *adxl345) enableInterrupts(ctx context.Context) error {
+	register := getByteFromInterrupts(adxl.interruptsEnabled)
+	return adxl.writeByte(ctx, IntEnable, register)
+}
+
+func (adxl *adxl345) setRelevantRegisters(ctx context.Context) error {
+	if len(adxl.interruptsEnabled) == 0 {
+		return nil
+	}
+	relevantRegisters := adxl.getRelevantRegisterValues()
+	for key := range defaultRegisterValues {
+		val, ok := relevantRegisters[key]
+		if ok {
+			err := adxl.writeByte(ctx, key, val)
+			if err != nil {
+				return errors.Errorf("Unable to write byte at register %x", key)
+			}
+		} else {
+			err := adxl.writeByte(ctx, key, defaultRegisterValues[key])
```

Comment:
The only part that's different between the if and else sections is the value that is written. I'd make that clearer:
```go
for key, value := range defaultRegisterValues {
    nondefaultValue, ok := relevantRegisters[key]
    if ok {
        value = nondefaultValue
    }
    err := adxl.writeByte(ctx, key, value)
    if err != nil {
        return err
    }
}
```

Reply:
...oh! and when I wrote that sample code, I forgot that you can define variables only in the scope of the if statement:
```go
if nondefaultValue, ok := relevantRegisters[key]; ok {
    value = nondefaultValue
}
```

That way, `nondefaultValue` and `ok` have very small scopes (just 3 lines), and a human reading this code doesn't need to keep track of them later (which isn't so bad for this short function, but can become really difficult for long functions that don't limit the scope of their state well).

Reply:
much cleaner thank you for pointing this out!